File: src/main/java/nu/peg/slack/pt/di/factory/TransportApiFactory.java
package nu.peg.slack.pt.di.factory;

import nu.peg.slack.pt.api.transport.TransportApi;
import org.glassfish.hk2.api.Factory;

public class TransportApiFactory implements Factory<TransportApi> {
    @Override
    public TransportApi provide() {
        return new TransportApi();
    }

    @Override
    public void dispose(TransportApi instance) {

    }
}

File: src/main/kotlin/nu/peg/slack/pt/di/factory/TransportApiFactory.kt
package nu.peg.slack.pt.di.factory

import nu.peg.slack.pt.api.transport.TransportApi

import org.glassfish.hk2.api.Factory

class TransportApiFactory : Factory<TransportApi> {
    fun provide(): TransportApi {
        return TransportApi()
    }

    fun dispose(instance: TransportApi) {

    }
}
